examples/factory-contract-global/src/lib.rs

```diff
@@ -38,7 +38,7 @@ impl GlobalFactoryContract {
             .create_account()
             .transfer(env::attached_deposit())
             .add_full_access_key(env::signer_account_pk())
-            .deploy_global_contract(code_bytes)
+            .publish_contract(code_bytes, None)
```

Comment:
1. I don't like using Option to signal if the contract will be published by account id or by hash.
2. Some() must only receive the `current_account_id` or new account if the new account is created, but in reality it is not even passed to the host, so it only misleads developers getting them thinking that they may deploy it for a different account id here